### ISSUE TITLE
Do not strip external ticketing systems references from email replies

### ISSUE THREAD

**Issue by avagithub** [zammad/zammad#1134]
When emails bounce back and forth between Zammad and another helpdesk tool (case at hand: salesforce), each reply from Zammad creates a new ticket in the other party's ticketing tool, as Zammad strips the [ref::xxxxxx::ref] from message body as well as subject, thus the friendly ticketing tool cannot recognise the message and opens a new ticket instead of appending it to the open case at hand.

Suggestion: option to include the most recent message sent by the other party ("quote the last message" to have the [ref:] in the message body) or do intelligently not strip other ticket references from message subjects.

At time of submission, we are running the most current 1.5.x installed by DEB method using postfix for outgoing mails.

Some more info:

Sample subject from message received:
RE: [Ticket#20170523800002] blahblahblah    [ ref:_00R54Vlq5._5209563BTk6:ref ]

Sample subject from Zammad reply to this:
[Ticket#20170523800002] blahblahblah

Effect:
Thank you for submitting your support request  [ ref:_00P74Uox8._5142F1SgC8M:ref ]

**Comment by hanneshal** [comment]
Zammad does not strip external references from ticket subjects/body. 
I tired it using the master branch using OTRS and Zammad. I expl. used your reference. 

So, if you just hit reply, the subject is used 1:1. 

Can you reproduce this behaviour? You could send a test mail from salesforce to support@znuny.com referencing to this issue# 

Thanks

**Comment by avagithub** [comment]
Thanks @hanneshal for your reply. Your observations are correct, but don't fully reflect my issue. Possibly my explanation was not spot on, I will therefore try again and add information I could obtain in the meantime.

The process is as follows:
In Zammad, create E-Mail Ticket
Customer = support@vendor.com
Title = "Bug Report XYZ"
Text = description of bug XYZ, of issue XYZ or of inquiry XYZ.

Vendor is using Salesforce, from which we get an an automated reply on our e-mail sent out by Zammad to support@vendor.com. The reply contains the case number of their salesforce instance (eg. CAxxxxxxxx) in the body, as well as a reference string (eg. ref:_00R54Vlq5._5209563BTk6:ref) added to the subject as well to the body of the automated reply.

tl; dr: the automated reply reads as follows:
> Subject: [Ticket#20170523800002] Bug Report XYZ [ ref:_00R54Vlq5._5209563BTk6:ref ]
> Body:
> Thank you for submitting technical support request:
> Bug Report XYZ
> 
> Your case number is: CA000009726
> We appreciate the importance of this issue and hope to send a reply promptly. 
> You may also find an answer to your question on our on-line FAQ
> 
> Thank you for your cooperation and patience.
> 
> [ ref:_00R54Vlq5._5209563BTk6:ref ]

Any mails sent to Zammad by the vendors Salesforce instance, be it automatically or by an agent working on the case, still have the Ticket#20170523800002 information in the subject and thus are correctly associated with a ticket in Zammad.
But _when we reply_ to one of their agents messages, the message Zammad sends out does neither contain their CAxxxxx nor ref:_xxxxxx strings neither in the body nor the subject, and thus opens a new ticket in their Salesforce instance.

Until now, we worked around this "limitation" by manually editing the title of a ticket in Zammad, adding the ref:_xxxxxx obtained by the automated reply to it by Salesforce, so any future mails sent out by Zammad have "[Ticket#20170523800002] Bug Report XYZ [ ref:_00R54Vlq5._5209563BTk6:ref ]" as their subject, thus Zammad recognizes it's own ticket as well as the vendors Salesforce their own ticket.

What I'm proposing:
When working on a ticket, when hitting reply on an article either _(full-)quote the article one is replying to in the body_ of the message Zammad sends out, or at least _use the subject of the article_, which in the end once was an e-mail Zammad received, _as the subject of the message Zammad sends out_.

Hope that makes more sense now.

**Comment by hanneshal** [comment]
OK, now I understand.

I changed the title of this issue to reflect your case. 
I see several soultions:
1) We crate the possibility to add gather external references (and store them in an object field) via Filter out of the ticket.
So we could merge them, if this is not automatically be done. You could search for it 
and you could use it in templates as a placeholder
2) We tell Zammad to use the last customer subject, not the ticket title.

Booth are imho very useful.

we come back to this. Thank you for the input.

**Comment by hiaz** [comment]
We also have the same problem. We talk (email) a lot with other IT companies and run very often into this issue. We tell them to use zammad instead of XYZ. But this is not always an option ;).

**Comment by NetBLOKS** [comment]
Hi,

we got the same problem, as we are communicating with providers etc. on behalf of our customers.
The Ticket ID i.e. from HostEurope is removed from the subject.
Would be wonderful if we could use the latest subject, just like the providers do (they answer to our Zammad-Mail and add the TicketID).
Maybe by choice (standard -> use Ticket Name, activate button -> Use new Subject)
In advance, it would be nice to view the mail sourcecode by clicking on a button, since the mail is only in Zammad.

Best regards!
